### DISCUSSION TITLE
I'm losing react TypeScript local component states (useState hook)

### DISCUSSION THREAD

**Discussion by mduvernon** [GrapesJS/grapesjs#5155] [Q&A]
I'm losing react TypeScript local component states (useState hook)

I'm trying to sync data (html and css) while using GrapesJS 
This sync process use event dispatcher/handler

On each update (or sync), I'm losing the previous state of my component object

It force me to use StateManager as a singleton class (with states store)

Can you please find a way to allow sync without having to use the StateManager ?

**Comment by artf** [comment]
Please provide a reproducible demo of your issue
